internal/app/marshaler.go

```diff
@@ -0,0 +1,40 @@
+package app
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	proto "google.golang.org/protobuf/proto"
+)
+
+// Marshaler describes anything that can marshal and unmarshal proto messages.
+type Marshaler interface {
```

Comment:
nit: no need to export this

Also, since it also unmarshals, this is typically named `codec`.

Reply:
Ah good call on both.